resources/files/windows/PuppetBolt/PuppetBoltFunctions.ps1

```diff
@@ -0,0 +1,1639 @@
+        [string]  $Query,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltRerunTypes]  $Rerun,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [switch] $Noop,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Description,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SSL=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SSLVerify=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $HostKeyCheck=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Username,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Password,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SSHPrivateKey,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $RunAs,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $SudoPassword,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $Concurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $CompileConcurrency,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ModulePath,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $BoltDir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $ConfigFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $InventoryFile,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $SaveRerun=$false,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Transport,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [int]     $ConnectTimeout,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $TTY=$false,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [string]  $Tmpdir,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [bool]    $Color=$true,
+
+        [parameter(ValueFromPipeline=$True,ValueFromPipelineByPropertyName)]
+        [BoltOutputLevel]  $OutputLevel
+    )
+    begin {
+        $_flags = Convert-BoltParams @PSBoundParameters
+    }
+    process {
+        return Invoke-BoltInternal -CommandString "script run $ScriptLocation" @PSBoundParameters
+    }
+}
+
+function Read-BoltPlans {
```

Comment:
Cmdlets should not be plural.  Prefer  `Read-BoltPlan`.  Note that the expected behaviour in PowerShell would be to output plan per object in the pipeline.  Not just spew json out as one object

Reply:
Actually, in retrospect this should really be `Get-BoltPlan` not read

https://docs.microsoft.com/en-us/powershell/developer/cmdlet/approved-verbs-for-windows-powershell-commands#get-vs-read